product_contract/views/product_template.xml

```diff
@@ -24,7 +24,7 @@
                       name="contract"
                       attrs="{'invisible': [('is_contract', '=', False)],}">
                     <group>
-                        <field name="contract_template_id"
+                        <field name="property_contract_template_id"
                                attrs="{'required':[('is_contract', '=', True)]}"/>
```

Comment:
I don't think this should be made required in multi-company context. Instead, raise an exception while creating the contract on sales order validation if not found.